Something went wrong. It is possible that you are trying to use an invalid token.

Hi!

When I try to enroll my token I'm getting this error message in every major browser.

<img width="982" alt="Screenshot 2021-11-19 at 14 55 38" src="https://user-images.githubusercontent.com/3431401/142635229-1b8bd131-958a-4c8d-aefd-c4f3459fa06f.png">

* SimpleSAMLphp: `v1.19.3`
* simplesamlphp/simplesamlphp-module-webauthn: `v0.11.0`
* The `usenewui` is enabled .
* MySQL database is configured and accessible.
* Debug mode is enables in the SSP and in the webauthn modul configuration.

My Yuibkey infos:

```
Device type: YubiKey 5Ci
Serial number: REDACTED
Firmware version: 5.2.4
Form factor: Keychain (USB-C, Lightning)
Enabled USB interfaces: OTP, FIDO, CCID

Applications
FIDO2       	Enabled
OTP         	Enabled
FIDO U2F    	Enabled
OATH        	Enabled
YubiHSM Auth	Not available
OpenPGP     	Enabled
PIV         	Enabled
```

This is what's in the SSP log (no php error so far):

```
Nov 19 14:03:33 idp-dev simplesamlphp[6722]: 7 [5c96f5f231] webauthn: userid: REDACTED
Nov 19 14:03:33 idp-dev simplesamlphp[6722]: 7 [5c96f5f231] User does not exist in DB, returning desired default.
Nov 19 14:03:33 idp-dev simplesamlphp[6722]: 7 [5c96f5f231] Saved state: '_8ab15a1c5dc6b97464b2d561e2bc1a09ecf6bedd85:https://idp.dev.einfra.hu/simplesaml/saml2/idp/SSOService.php?spentityid=https.eduid.hu&RelayState=ss&cookieTime=1637330608'
Nov 19 14:03:33 idp-dev simplesamlphp[6721]: 6 [5c96f5f231] FIDO2 - Accessing WebAuthn interface
Nov 19 14:03:33 idp-dev simplesamlphp[6721]: 7 [5c96f5f231] Loading state: '_8ab15a1c5dc6b97464b2d561e2bc1a09ecf6bedd85:https://idp.dev.einfra.hu/simplesaml/saml2/idp/SSOService.php?spentityid=https.eduid.hu&RelayState=ss&cookieTime=1637330608'
Nov 19 14:03:33 idp-dev simplesamlphp[6721]: 7 [5c96f5f231] Localization: load domain 'messages' at '/srv/simplesamlphp/locales'
Nov 19 14:03:33 idp-dev simplesamlphp[6721]: 7 [5c96f5f231] Trying langpath for 'en' as '/srv/simplesamlphp/locales/en/LC_MESSAGES/'
Nov 19 14:03:33 idp-dev simplesamlphp[6721]: 7 [5c96f5f231] Localization: load domain 'webauthn' at '/srv/simplesamlphp/modules/webauthn/locales'
Nov 19 14:03:33 idp-dev simplesamlphp[6721]: 7 [5c96f5f231] Trying langpath for 'en' as '/srv/simplesamlphp/modules/webauthn/locales/en/LC_MESSAGES/'
```

If I close the dialog window and hit enter:

```
Nov 19 14:12:22 idp-dev simplesamlphp[6722]: 6 [5c96f5f231] FIDO2 - Accessing WebAuthn enrollment validation
Nov 19 14:12:22 idp-dev simplesamlphp[6722]: 7 [5c96f5f231] Loading state: '_8ab15a1c5dc6b97464b2d561e2bc1a09ecf6bedd85:https://idp.dev.einfra.hu/simplesaml/saml2/idp/SSOService.php?spentityid=https.eduid.hu&RelayState=ss&cookieTime=1637330608'
Nov 19 14:12:22 idp-dev simplesamlphp[6722]: 3 [5c96f5f231] SimpleSAML\Error\Error: UNHANDLEDEXCEPTION
Nov 19 14:12:22 idp-dev simplesamlphp[6722]: 3 [5c96f5f231] Backtrace:
Nov 19 14:12:22 idp-dev simplesamlphp[6722]: 3 [5c96f5f231] 2 /srv/simplesamlphp/www/_include.php:17 (SimpleSAML_exception_handler)
Nov 19 14:12:22 idp-dev simplesamlphp[6722]: 3 [5c96f5f231] 1 /srv/simplesamlphp/vendor/symfony/error-handler/ErrorHandler.php:607 (Symfony\Component\ErrorHandler\ErrorHandler::handleException)
Nov 19 14:12:22 idp-dev simplesamlphp[6722]: 3 [5c96f5f231] 0 [builtin] (N/A)
Nov 19 14:12:22 idp-dev simplesamlphp[6722]: 3 [5c96f5f231] Caused by: InvalidArgumentException: Cannot parse the data. Found invalid Additional Information "11110" (30).
Nov 19 14:12:22 idp-dev simplesamlphp[6722]: 3 [5c96f5f231] Backtrace:
Nov 19 14:12:22 idp-dev simplesamlphp[6722]: 3 [5c96f5f231] 9 /srv/simplesamlphp/vendor/spomky-labs/cbor-php/src/Decoder.php:62 (CBOR\Decoder::process)
Nov 19 14:12:22 idp-dev simplesamlphp[6722]: 3 [5c96f5f231] 8 /srv/simplesamlphp/vendor/spomky-labs/cbor-php/src/Decoder.php:89 (CBOR\Decoder::processFinite)
Nov 19 14:12:22 idp-dev simplesamlphp[6722]: 3 [5c96f5f231] 7 /srv/simplesamlphp/vendor/spomky-labs/cbor-php/src/Decoder.php:67 (CBOR\Decoder::process)
Nov 19 14:12:22 idp-dev simplesamlphp[6722]: 3 [5c96f5f231] 6 /srv/simplesamlphp/vendor/spomky-labs/cbor-php/src/Decoder.php:43 (CBOR\Decoder::decode)
Nov 19 14:12:22 idp-dev simplesamlphp[6722]: 3 [5c96f5f231] 5 /srv/simplesamlphp/modules/webauthn/lib/WebAuthn/WebAuthnAbstractEvent.php:413 (SimpleSAML\Module\webauthn\WebAuthn\WebAuthnAbstractEvent::cborDecode)
Nov 19 14:12:22 idp-dev simplesamlphp[6722]: 3 [5c96f5f231] 4 /srv/simplesamlphp/modules/webauthn/lib/WebAuthn/WebAuthnRegistrationEvent.php:76 (SimpleSAML\Module\webauthn\WebAuthn\WebAuthnRegistrationEvent::__construct)
Nov 19 14:12:22 idp-dev simplesamlphp[6722]: 3 [5c96f5f231] 3 /srv/simplesamlphp/modules/webauthn/lib/Controller/RegProcess.php:124 (SimpleSAML\Module\webauthn\Controller\RegProcess::main)
Nov 19 14:12:22 idp-dev simplesamlphp[6722]: 3 [5c96f5f231] 2 /srv/simplesamlphp/modules/webauthn/www/regprocess.php:14 (require)
Nov 19 14:12:22 idp-dev simplesamlphp[6722]: 3 [5c96f5f231] 1 /srv/simplesamlphp/lib/SimpleSAML/Module.php:266 (SimpleSAML\Module::process)
Nov 19 14:12:22 idp-dev simplesamlphp[6722]: 3 [5c96f5f231] 0 /srv/simplesamlphp/www/module.php:10 (N/A)
Nov 19 14:12:22 idp-dev simplesamlphp[6722]: 3 [5c96f5f231] Error report with id 2492a410 generated.
Nov 19 14:12:22 idp-dev simplesamlphp[6722]: 7 [5c96f5f231] Session: Valid session found with 'niifp-ldap'.
Nov 19 14:12:22 idp-dev simplesamlphp[6722]: 7 [5c96f5f231] Localization: load domain 'messages' at '/srv/simplesamlphp/locales'
Nov 19 14:12:22 idp-dev simplesamlphp[6722]: 7 [5c96f5f231] Trying langpath for 'en' as '/srv/simplesamlphp/locales/en/LC_MESSAGES/'
```

> And please confirm that your web server with SSP is actually running on a FQDN within 'eduid.hu'. Otherwise, the transaction will rightly be refused by the browser.

It was not on the correct FQDN (einfra.hu), so I changed it and the token registration started.

After completing the registration, this error message comes up:

```
Nov 19 14:28:44 idp-dev simplesamlphp[6722]: 7 [8fb1932cd9] webauthn: userid: REDACTED
Nov 19 14:28:44 idp-dev simplesamlphp[6722]: 7 [8fb1932cd9] User does not exist in DB, returning desired default.
Nov 19 14:28:44 idp-dev simplesamlphp[6722]: 7 [8fb1932cd9] Saved state: '_fe8e1a9370199031ede2201cf6dd200c5e1e00751c:https://idp.dev.einfra.hu/simplesaml/saml2/idp/SSOService.php?spentityid=https.eduid.hu&RelayState=ss&cookieTime=1637332121'
Nov 19 14:28:44 idp-dev simplesamlphp[6721]: 6 [8fb1932cd9] FIDO2 - Accessing WebAuthn interface
Nov 19 14:28:44 idp-dev simplesamlphp[6721]: 7 [8fb1932cd9] Loading state: '_fe8e1a9370199031ede2201cf6dd200c5e1e00751c:https://idp.dev.einfra.hu/simplesaml/saml2/idp/SSOService.php?spentityid=https.eduid.hu&RelayState=ss&cookieTime=1637332121'
Nov 19 14:28:44 idp-dev simplesamlphp[6721]: 7 [8fb1932cd9] Localization: load domain 'messages' at '/srv/simplesamlphp/locales'
Nov 19 14:28:44 idp-dev simplesamlphp[6721]: 7 [8fb1932cd9] Trying langpath for 'en' as '/srv/simplesamlphp/locales/en/LC_MESSAGES/'
Nov 19 14:28:44 idp-dev simplesamlphp[6721]: 7 [8fb1932cd9] Localization: load domain 'webauthn' at '/srv/simplesamlphp/modules/webauthn/locales'
Nov 19 14:28:44 idp-dev simplesamlphp[6721]: 7 [8fb1932cd9] Trying langpath for 'en' as '/srv/simplesamlphp/modules/webauthn/locales/en/LC_MESSAGES/'
Nov 19 14:29:04 idp-dev simplesamlphp[6721]: 6 [8fb1932cd9] FIDO2 - Accessing WebAuthn enrollment validation
Nov 19 14:29:04 idp-dev simplesamlphp[6721]: 7 [8fb1932cd9] Loading state: '_fe8e1a9370199031ede2201cf6dd200c5e1e00751c:https://idp.dev.einfra.hu/simplesaml/saml2/idp/SSOService.php?spentityid=https.eduid.hu&RelayState=ss&cookieTime=1637332121'
Nov 19 14:29:04 idp-dev simplesamlphp[6721]: 4 [8fb1932cd9] Missing AAGUID configuration file (/srv/simplesamlphp/config/webauthn-aaguid.json). No device will be recognized.
Nov 19 14:29:04 idp-dev simplesamlphp[6721]: 6 [8fb1932cd9] AAGUID c5ef55ffad9a4b9fb580adebafe026d0 not found in dictionary, device is unknown.
Nov 19 14:29:04 idp-dev simplesamlphp[6721]: 7 [8fb1932cd9] Credential does not exist yet.
Nov 19 14:29:04 idp-dev simplesamlphp[6721]: 6 [8fb1932cd9] AAGUID c5ef55ffad9a4b9fb580adebafe026d0 not found in dictionary, device is unknown.
Nov 19 14:29:04 idp-dev simplesamlphp[6721]: 7 [8fb1932cd9] Saved state: '_fe8e1a9370199031ede2201cf6dd200c5e1e00751c:https://idp.dev.einfra.hu/simplesaml/saml2/idp/SSOService.php?spentityid=https.eduid.hu&RelayState=ss&cookieTime=1637332121'
Nov 19 14:29:04 idp-dev simplesamlphp[6721]: 3 [8fb1932cd9] SimpleSAML\Error\Error: UNHANDLEDEXCEPTION
Nov 19 14:29:04 idp-dev simplesamlphp[6721]: 3 [8fb1932cd9] Backtrace:
Nov 19 14:29:04 idp-dev simplesamlphp[6721]: 3 [8fb1932cd9] 2 /srv/simplesamlphp/www/_include.php:20 (SimpleSAML_exception_handler)
Nov 19 14:29:04 idp-dev simplesamlphp[6721]: 3 [8fb1932cd9] 1 /srv/simplesamlphp/vendor/symfony/error-handler/ErrorHandler.php:607 (Symfony\Component\ErrorHandler\ErrorHandler::handleException)
Nov 19 14:29:04 idp-dev simplesamlphp[6721]: 3 [8fb1932cd9] 0 [builtin] (N/A)
Nov 19 14:29:04 idp-dev simplesamlphp[6721]: 3 [8fb1932cd9] Caused by: ArgumentCountError: Too few arguments to function SimpleSAML\Module\webauthn\Controller\RegProcess::SimpleSAML\Module\webauthn\Controller\{closure}(), 0 passed in /srv/simplesamlphp/vendor/symfony/http-foundation/StreamedResponse.php on line 109 and exactly 2 expected
Nov 19 14:29:04 idp-dev simplesamlphp[6721]: 3 [8fb1932cd9] Backtrace:
Nov 19 14:29:04 idp-dev simplesamlphp[6721]: 3 [8fb1932cd9] 5 /srv/simplesamlphp/modules/webauthn/lib/Controller/RegProcess.php:200 (SimpleSAML\Module\webauthn\Controller\RegProcess::SimpleSAML\Module\webauthn\Controller\{closure})
Nov 19 14:29:04 idp-dev simplesamlphp[6721]: 3 [8fb1932cd9] 4 /srv/simplesamlphp/vendor/symfony/http-foundation/StreamedResponse.php:109 (Symfony\Component\HttpFoundation\StreamedResponse::sendContent)
Nov 19 14:29:04 idp-dev simplesamlphp[6721]: 3 [8fb1932cd9] 3 /srv/simplesamlphp/vendor/symfony/http-foundation/Response.php:394 (Symfony\Component\HttpFoundation\Response::send)
Nov 19 14:29:04 idp-dev simplesamlphp[6721]: 3 [8fb1932cd9] 2 /srv/simplesamlphp/modules/webauthn/www/regprocess.php:15 (require)
Nov 19 14:29:04 idp-dev simplesamlphp[6721]: 3 [8fb1932cd9] 1 /srv/simplesamlphp/lib/SimpleSAML/Module.php:266 (SimpleSAML\Module::process)
Nov 19 14:29:04 idp-dev simplesamlphp[6721]: 3 [8fb1932cd9] 0 /srv/simplesamlphp/www/module.php:10 (N/A)
Nov 19 14:29:04 idp-dev simplesamlphp[6721]: 3 [8fb1932cd9] Error report with id e85b23c3 generated.
Nov 19 14:29:04 idp-dev simplesamlphp[6721]: 7 [8fb1932cd9] Session: Valid session found with 'niifp-ldap'.
Nov 19 14:29:04 idp-dev simplesamlphp[6721]: 7 [8fb1932cd9] Localization: load domain 'messages' at '/srv/simplesamlphp/locales'
Nov 19 14:29:04 idp-dev simplesamlphp[6721]: 7 [8fb1932cd9] Trying langpath for 'en' as '/srv/simplesamlphp/locales/en/LC_MESSAGES/'
```

> This concrete error is raised by a third-party library. What is the version of spomky-labs/cbor-php on your deployment?

```
"spomky-labs/cbor-php": "^1.0"
```

I tried with it, but I got the same error:

```
Nov 19 15:04:06 idp-dev simplesamlphp[6721]: 3 [88005c9d16] Caused by: ArgumentCountError: Too few arguments to function SimpleSAML\Module\webauthn\Controller\AuthProcess::SimpleSAML\Module\webauthn\Controller\{closure}(), 0 passed in /srv/simplesamlphp/vendor/symfony/http-foundation/StreamedResponse.php on line 109 and exactly 2 expected
Nov 19 15:04:06 idp-dev simplesamlphp[6721]: 3 [88005c9d16] Backtrace:
Nov 19 15:04:06 idp-dev simplesamlphp[6721]: 3 [88005c9d16] 5 /srv/simplesamlphp/modules/webauthn/lib/Controller/AuthProcess.php:181 (SimpleSAML\Module\webauthn\Controller\AuthProcess::SimpleSAML\Module\webauthn\Controller\{closure})
Nov 19 15:04:06 idp-dev simplesamlphp[6721]: 3 [88005c9d16] 4 /srv/simplesamlphp/vendor/symfony/http-foundation/StreamedResponse.php:109 (Symfony\Component\HttpFoundation\StreamedResponse::sendContent)
Nov 19 15:04:06 idp-dev simplesamlphp[6721]: 3 [88005c9d16] 3 /srv/simplesamlphp/vendor/symfony/http-foundation/Response.php:394 (Symfony\Component\HttpFoundation\Response::send)
Nov 19 15:04:06 idp-dev simplesamlphp[6721]: 3 [88005c9d16] 2 /srv/simplesamlphp/modules/webauthn/www/authprocess.php:15 (require)
Nov 19 15:04:06 idp-dev simplesamlphp[6721]: 3 [88005c9d16] 1 /srv/simplesamlphp/lib/SimpleSAML/Module.php:266 (SimpleSAML\Module::process)
Nov 19 15:04:06 idp-dev simplesamlphp[6721]: 3 [88005c9d16] 0 /srv/simplesamlphp/www/module.php:10 (N/A)
```


* `https://raw.githubusercontent.com/simplesamlphp/simplesamlphp-module-webauthn/b6b532948b110f586784fd4675f8075850a15512/lib/Controller/RegProcess.php`

--- a/lib/Controller/AuthProcess.php
+++ b/lib/Controller/AuthProcess.php
@@ -17,7 +17,6 @@ use SimpleSAML\XHTML\Template;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\RedirectResponse;
 use Symfony\Component\HttpFoundation\Response;
-use Symfony\Component\HttpFoundation\StreamedResponse;
 
 /**
  * Controller class for the webauthn module.
@@ -91,8 +90,7 @@ class AuthProcess
     /**
      * @param \Symfony\Component\HttpFoundation\Request $request
      * @return (\Symfony\Component\HttpFoundation\RedirectResponse|
-     *         \SimpleSAML\HTTP\RunnableResponse|
-     *         \Symfony\Component\HttpFoundation\StreamedResponse)  A Symfony Response-object.
+     *         \SimpleSAML\HTTP\RunnableResponse)  A Symfony Response-object.
      */
     public function main(Request $request): Response
     {
@@ -173,12 +171,14 @@ class AuthProcess
         $this->authState::saveState($state, 'webauthn:request');
 
         if ($debugEnabled) {
-            $response = new StreamedResponse();
-            $response->setCallback(function ($authObject, $state) {
-                echo $authObject->getDebugBuffer();
-                echo $authObject->getValidateBuffer();
-                echo "Debug mode, not continuing to " . ($state['FIDO2WantsRegister'] ? "credential registration page." : "destination.");
-            });
+            $response = new RunnableResponse(
+                function ($authObject, $state) {
+                    echo $authObject->getDebugBuffer();
+                    echo $authObject->getValidateBuffer();
+                    echo "Debug mode, not continuing to " . ($state['FIDO2WantsRegister'] ? "credential registration page." : "destination.");
+                },
+                [$authObject, $state]
+            );
         } else {
             if ($state['FIDO2WantsRegister']) {
                 $response = new RedirectResponse(Module::getModuleURL('webauthn/webauthn?StateId=' . urlencode($stateId)));

--- a/lib/Controller/RegProcess.php
+++ b/lib/Controller/RegProcess.php
@@ -92,7 +92,7 @@ class RegProcess
 
     /**
      * @param \Symfony\Component\HttpFoundation\Request $request
-     * @return \Symfony\Component\HttpFoundation\RedirectResponse|\SimpleSAML\HTTP\RunnableResponse|\Symfony\Component\HttpFoundation\StreamedResponse
+     * @return \Symfony\Component\HttpFoundation\RedirectResponse|\SimpleSAML\HTTP\RunnableResponse
      *   A Symfony Response-object.
      */
     public function main(Request $request): Response

--- a/lib/Controller/RegProcess.php
+++ b/lib/Controller/RegProcess.php
@@ -18,7 +18,8 @@ use SimpleSAML\XHTML\Template;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\RedirectResponse;
 use Symfony\Component\HttpFoundation\Response;
-use Symfony\Component\HttpFoundation\StreamedResponse;
+
+use function urlencode;
 
 /**
  * Controller class for the webauthn module.
@@ -194,14 +195,16 @@ class RegProcess
 
         $id = $this->authState::saveState($state, 'webauthn:request');
         if ($debugEnabled === true) {
-            $response = new StreamedResponse();
-            $response->setCallback(function ($regObject, $id) {
-                echo $regObject->getDebugBuffer();
-                echo $regObject->getValidateBuffer();
-                echo "<form id='regform' method='POST' action='" .
-                    Module::getModuleURL('webauthn/webauthn?StateId=' . urlencode($id)) . "'>";
-                echo "<button type='submit'>Return to previous page.</button>";
-            });
+            $response = new RunnableResponse(
+                function ($regObject, $id) {
+                    echo $regObject->getDebugBuffer();
+                    echo $regObject->getValidateBuffer();
+                    echo "<form id='regform' method='POST' action='" .
+                        Module::getModuleURL('webauthn/webauthn?StateId=' . urlencode($id)) . "'>";
+                    echo "<button type='submit'>Return to previous page.</button>";
+                },
+                [$regObject, $id]
+            );
         } elseif (array_key_exists('Registration', $state)) {
             $response = new RedirectResponse(Module::getModuleURL('webauthn/webauthn?StateId=' . urlencode($id)));
         } else {

